e2store/src/era2.rs

```diff
@@ -139,8 +127,51 @@ impl Era2 {
     }
 }
 
+impl Iterator for Era2 {
+    type Item = AccountOrStorageEntry;
```

Comment:
I think it's better for `Item` to be `anyhow::Result<AccountOrStorageEntry>`.

I personally really don't like using `panic!`, because caller can no longer handle the failure.
And the `Era2` structure seems more like a library, so my feeling is that it should allow caller to handle errors.